### PR TITLE
[CDAP-16567] Properly fix styling for plugin configuration title

### PR DESCRIPTION
**Pull request by ajainarayanan**
- We made minor modification to the plugin configuration title change but it was incomplete and never worked correctly. (https://github.com/cdapio/cdap/pull/11879).
- This is to properly fix the change 

JIRA: https://issues.cask.co/browse/CDAP-16567
Build: https://builds.cask.co/browse/CDAP-UDUT567-3

### CODE REVIEW

File: cdap-ui/app/cdap/components/ConfigurationGroup/index.tsx [hunk 1]
```diff
@@ -38,13 +38,16 @@ const styles = (theme): StyleRules => {
     },
     groupTitle: {
       marginBottom: '15px',
+      marginLeft: 10,
+      marginRight: 15,
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/ConfigurationGroup/index.tsx]: nit: make it consistent and have `px`

File: cdap-ui/app/cdap/components/ConfigurationGroup/index.tsx [hunk 2]
```diff
@@ -38,13 +38,16 @@ const styles = (theme): StyleRules => {
...
     },
     h2Title: {
-      ...h2Styles(theme),
-      marginBottom: 0,
+      ...h2Styles(theme).root,
+      marginBottom: 5,
     },
     groupSubTitle: {
       color: theme.palette.grey[200],
+      display: 'inline-block',
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/ConfigurationGroup/index.tsx]: why is this needed? When I see in the studio, the description is under the group title since the title is a block element.

**Review by elfenheart — APPROVED** [review]
👍 LGTM

**Comment by yukiej** [comment]
lgtm!
